### PR TITLE
BE-8167: List previous test results

### PR DESCRIPTION
**Pull request by ajanata**
Add enumerate results to db interface. Add endpoint to list them, using
an HTML template.

Needs #12.

### CODE REVIEW

File: templates/result_list.gohtml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<!DOCTYPE html>
```
Comment (ajanata) [inline templates/result_list.gohtml:1]: extension is to indicate to goland that go templating applies to this file

File: templates/result_list.gohtml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            <thead class="thead-default">
+                <tr>
+                    <th scope="col">Start Time</th>
+                    <th scope="col">Duration</th>
+                    <th scope="col">Total Tests Executed</th>
+                    <th scope="col">Tests Passed</th>
+                </tr>
+            </thead>
+            <tbody>
+            {{- /*gotype: github.com/gametimesf/testy.listResultsCtx*/ -}}
```
Comment (ajanata) [inline templates/result_list.gohtml:21]: goland uses this to know what to autocomplete with

File: db.go [hunk 3]
```diff
@@ -12,6 +13,16 @@ var ErrNoDB = errors.New("no DB set")
...
+	// Enumerate lists the test results for the given page. The datastore determines the page size.
+	Enumerate(ctx context.Context, page int) (results []Summary, more bool, err error)
+}
+
+type Summary struct {
+	ID      string
+	Started time.Time
+	Dur     time.Duration
+	Total   int
+	Passed  int
```
Comment (jeffatgametime) [inline db.go:25]: gonna make us do math to know how many failed?!?!

File: templates/result_list.gohtml (hunk 2 above)
Comment (jeffatgametime) [reply templates/result_list.gohtml:21]: oh wow, of course TIL

**Review by jeffatgametime — APPROVED** [review]

File: db.go (hunk 3 above)
Comment (waitstime) [reply db.go:25]: I would prefer failed instead of pass if had to choose.

**Review by waitstime — APPROVED** [review]
Would prefer Failed instead of Passed since it's always expected to be 0 instead of unknown for pass. Or show both

File: db.go (hunk 3 above)
Comment (Seanskiver) [reply db.go:25]: Why not both?

**Review by Seanskiver — APPROVED** [review]

**Comment by ajanata** [comment]
I'd like to request that any visual changes (including adding a failed column) be done all at once after the primary PRs.

**Pull request merged**